src/ng/compile.js

```diff
@@ -2592,11 +2587,13 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '=':
-            if (optional && !attrs[attrName]) {
-              return;
+            if (!hasOwnProperty.call(attrs, attrName)) {
```

Comment:
Oh, the example snippet in that issue also handles values which are exclusively whitespace, but I think these are pretty rare